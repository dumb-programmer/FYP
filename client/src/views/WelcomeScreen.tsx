export default function WelcomeScreen() {
    return <div className="h-full w-full flex flex-col gap-4 items-center justify-center" data-testid="welcome-screen">
        <h1 className="text-5xl font-bold">Welcome</h1>
        <p className="text-gray-500">Click on create chat or on existing chat to start chatting</p>
    </div>;
}