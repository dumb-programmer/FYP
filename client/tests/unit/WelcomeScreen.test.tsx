import React from "react";
import { render, screen } from "@testing-library/react";
import { expect, describe, it } from "vitest";
import WelcomeScreen from "../../src/views/WelcomeScreen";

const renderComponent = () => render(<WelcomeScreen />);

describe("WelcomeScreen", () => {
    it("renders correctly", () => {
        renderComponent();

        const welcomeScreen = screen.getByTestId("welcome-screen");

        expect(welcomeScreen).toMatchSnapshot();
    });
})