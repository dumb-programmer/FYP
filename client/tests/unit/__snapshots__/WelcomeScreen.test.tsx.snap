// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WelcomeScreen > renders correctly 1`] = `
<div
  class="h-full w-full flex flex-col gap-4 items-center justify-center"
  data-testid="welcome-screen"
>
  <h1
    class="text-5xl font-bold"
  >
    Welcome
  </h1>
  <p
    class="text-gray-500"
  >
    Click on create chat or on existing chat to start chatting
  </p>
</div>
`;
